src/engine.ts

```diff
@@ -517,6 +577,12 @@ export function initializePolyfill() {
           for (const child of node.childNodes) {
             getOrCreateInstance(child).update(currentState);
           }
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const shadow = (node as any)[SHADOW_SYMBOL];
+          if (shadow) {
```

Comment:
We should just set this as the `stateProvider` at the same time we set the `innerController`

Reply:
ahhh! this was the piece I was struggling to see, this makes sense.